pymbolic/parser.py

```diff
@@ -144,6 +144,8 @@ class Parser:
                     r"[0-9]*\.[0-9]+[eEdD][+-]?[0-9]+([a-zA-Z]*)\b"),
                 # has a letter tag
                 pytools.lex.RE(r"[0-9]+([a-zA-Z]+)"),
+                # inf, nan
+                pytools.lex.RE("inf"), pytools.lex.RE("nan"),
```

Comment:
This isn't quite what Python does. `nan` is the identifier, not a floating point number. I do agree that we want literals for these guys, but I'm not totally sure I like these. Python's `float("nan")` seems a bit out of character for (e.g.) loopy. Any other ideas?

Reply:
1. The other possibility is to provide a namespace for such keywords, but that would violate
```python
assert print(parse("inf")) == "inf"
```

2. We could also make the parser take a flag to treat these as keywords or otherwise.

Reply:
What's wrong with letting user code run a mapper to run symbol lookup on these things? I think in loopy lang we will ultimately want to allow `from numpy import inf`, to provide a principled way of accessing these things. (To be clear, I am only thinking of data and `LoopKernel` objects as being potentially importable.) But doing so would require loopy to remain in charge of symbol lookup.

I'm OK with adding a flag for the behavior you propose though.